File: register_with_eureka.py
from py_eureka_client import eureka_client

async def register_with_eureka(app_name, port):
    await eureka_client.init_async(
        eureka_server="http://localhost:8761/eureka/",
        app_name=app_name.upper(),
        instance_port=port,
        instance_host="localhost",
        instance_ip="127.0.0.1",
        health_check_url=f"http://localhost:{port}/transaction/health",
        status_page_url=f"http://localhost:{port}/transaction/health"
    )

File: registerWithEureka.js
const { Eureka } = require("eureka-js-client");

function registerWithEureka(appName, PORT) {
  const client = new Eureka({
    instance: {
      app: appName.toUpperCase(),
      hostName: "localhost",
      ipAddr: "127.0.0.1",
      statusPageUrl: `http://localhost:${PORT}/service/health`,
      port: {
        "$": PORT,
        "@enabled": true
      },
      vipAddress: appName,
      dataCenterInfo: {
        "@class": "com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo",
        name: "MyOwn"
      }
    },
    eureka: {
      host: "localhost",
      port: 8761,
      servicePath: "/eureka/apps/"
    }
  });

  client.logger.level("debug");
  client.start((error) => {
    if (error) {
      console.error("Eureka registration failed:", error);
    } else {
      console.log("Eureka registration complete");
    }
  });
}

module.exports = registerWithEureka;
